src/python_testing/TC_RVCCLEANM_1_2.py

```diff
@@ -55,64 +69,74 @@ async def test_TC_RVCCLEANM_1_2(self):
 
             logging.info("SupportedModes: %s" % (supported_modes))
 
+            # Verify that the list has at least 2 and at most 255 entries
             asserts.assert_greater_equal(len(supported_modes), 2, "SupportedModes must have at least 2 entries!")
             asserts.assert_less_equal(len(supported_modes), 255, "SupportedModes must have at most 255 entries!")
 
-            modes = []
+            # Verify that each ModeOptionsStruct entry has a unique Mode field value
```

Comment:
assert.assert_equal(len(set(supported_modes)), len(supported_modes))
self.supported_modes_dut = supported_modes

Reply:
`supported_modes` is a structure. Here we want to check that the values used in the `mode` field of this struct are unique. Using the `set` function on the list of structs would mean that two structs that use the same value for the `mode` field but have different entries for the `label` and `madeTags` entries would be unique. We don't want this.

Reply:
OH. Gotcha. that's why mode.mode is used. makes more sense now. KK, just close this then.